lib/turn.rb

```diff
@@ -0,0 +1,78 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war, :type, :winner
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+    @winner = winner
+  end
+
+  def type
+      if player1.deck.rank_of_card_at(0) != player2.deck.rank_of_card_at(0)
+        :basic
+      elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) != player2.deck.rank_of_card_at(2)
+        :war
+      elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+        :mutually_assured_destruction
+      end
+  end
+
+  def winner
+    if type == :basic
+      if player1.deck.rank_of_card_at(0) > player2.deck.rank_of_card_at(0)
+        @player1
+      else
+        @player2
+      end
+      #return the player with the highest ranking card at index position 0
+    elsif type == :war
+      if player1.deck.rank_of_card_at(2) > player2.deck.rank_of_card_at(2)
+        @player1
+      else
+        @player2
+      end
+    elsif type == :mutually_assured_destruction
+      "No Winner"
+    end
+  end
+
+  def pile_cards
+      if type == :basic
+      #each player will send one card (the top card) to the spoils pile
+      # spoils_of_war = []
+      spoils_of_war << player1.deck.remove_card
+      spoils_of_war << player2.deck.remove_card
+      # spoils_of_war
+      end
+
+      if type == :war
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+      end
+
+      if type == :mutually_assured_destruction
+        player1.deck.remove_card
+        player1.deck.remove_card
+        player1.deck.remove_card
+        player2.deck.remove_card
+        player2.deck.remove_card
+        player2.deck.remove_card
+      end
```

Comment:
The final 2 `if` blocks have quite a bit of repetition, what could we have done here instead to make this more DRY? Take a look at the `#times` method for clues.